code/modules/clothing/spacesuits/rig/rig.dm

```diff
@@ -913,6 +922,54 @@
 /mob/living/carbon/human/get_rig()
 	return back
 
+
+//Used in random rig spawning for cargo
+//Randomly deletes modules
+/obj/item/weapon/rig/proc/lose_modules(var/probability)
+	for(var/obj/item/rig_module/module in installed_modules)
+		if (probability)
+			qdel(module)
+
+
+//Fiddles with some wires to possibly make the suit malfunction a little
+//Had to use numeric literals here, the wire defines in rig_wiring.dm weren't working
+//Possibly due to being defined in a later file, or undef'd somewhere
+/obj/item/weapon/rig/proc/misconfigure(var/probability)
+	if (prob(probability))
+		wires.UpdatePulsed(RIG_SECURITY)//Fiddle with access
+	if (prob(probability))
+		wires.UpdatePulsed(RIG_AI_OVERRIDE)//frustrate the AI
+	if (prob(probability))
+		wires.UpdateCut(RIG_SYSTEM_CONTROL)//break the suit
+	if (prob(probability))
+		wires.UpdatePulsed(RIG_INTERFACE_LOCK)
+	if (prob(probability))
+		wires.UpdateCut(RIG_INTERFACE_SHOCK)
+	if (prob(probability))
+		subverted = 1
+
+//Drains, rigs or removes the cell
+/obj/item/weapon/rig/proc/sabotage_cell()
+	if (!cell)
+		return
+
+	if (prob(50))
+		cell.charge = rand(0, cell.charge*0.5)
+	else if (prob(15))
+		cell.rigged = 1
```

Comment:
That's fine, they can replace it if they can find a suitable one.
They're not likely to ask for help with it because rigs, being a rare treasure, would likely be siezed if an assistant mentioned he had one

Even if its a one time experience, the first time is memorable